src/plugins/flow.js

```diff
@@ -529,6 +552,7 @@ pp.flowParsePrimaryType = function () {
   let tmp;
   let type;
   let isGroupedType = false;
+  let oldNoAnonFunctionType = this.state.noAnonFunctionType;
```

Comment:
Can you please add `noAnonFunctionType` to the [state](https://github.com/babel/babylon/blob/master/src/tokenizer/state.js) so that v8 can better optimize the usage of the state.